libbeat/reader/readfile/timeout.go

```diff
@@ -67,6 +69,11 @@ func (r *TimeoutReader) Next() (reader.Message, error) {
 		r.running = true
 		go func() {
 			for {
+				select {
+				case <-r.done:
+					return
+				default:
+				}
 				message, err := r.reader.Next()
 				r.ch <- lineMessage{message, err}
```

Comment:
I think we need a select here as well.

Reply:
During stopping the harvester, all of the reader pipeline elements are closed. Thus, at one point, `Next` gets an EOF from the underlying multiline reader and the goroutine is able to stop. But to be 1000000% sure it won't block in the future, this won't be the default case anymore.

Reply:
How reader.Next returns us an error I can see. But it looks like is the `r.ch <- ...` line where we would deadlock on shutdown, potentially leaking the go-routine.